front/app/modules/commercial/project_folders/permissions/rules.ts

```diff
@@ -3,27 +3,30 @@ import {
   definePermissionRule,
   IRouteItem,
 } from 'services/permissions/permissions';
-import { isProjectFolderModerator, moderatesFolder } from './roles';
+import { isProjectFolderModerator, userModeratesFolder } from './roles';
 import { isAdmin } from 'services/permissions/roles';
 import {
   canAccessRoute,
-  isAdminRoute,
-  MODERATOR_ROUTES,
+  isModeratorRoute,
 } from 'services/permissions/rules/routePermissions';
 import { IUser } from 'services/users';
 import { IAppConfigurationData } from 'services/appConfiguration';
+import { isNilOrError } from 'utils/helperUtils';
 
 const canUserAccessAdminFolderRoute = (
   item: IRouteItem,
   user: IUser | null,
   tenant: IAppConfigurationData
 ) => {
-  return (
-    canAccessRoute(item, user, tenant) ||
-    (isAdminRoute(item.path) &&
-      (user ? isProjectFolderModerator(user.data) : false) &&
-      MODERATOR_ROUTES.includes(item.path))
-  );
+  const hasAdminFolderRouteAccess =
+    !isNilOrError(user) &&
+    isProjectFolderModerator(user.data) &&
+    // folder mods have the same
+    // access rights as project mods
+    // besides their respective folders/projects
+    (isModeratorRoute(item) || item.path.includes('folders'));
```

Comment:
I'm not sure how to quickly test this, but can't we make this a bit safer?

`
item.path.includes('folders')
`

 Won't any path with folders in it (now and new ones in the future) pass here?

Reply:
Good point 👍